actions/login.php

```diff
@@ -38,23 +37,20 @@
 
 $result = elgg_authenticate($username, $password);
 if ($result !== true) {
-	register_error($result);
-	forward(REFERER);
+	return elgg_error_response($result, REFERRER, ELGG_HTTP_FORBIDDEN);
```

Comment:
I don't know any site that considers incorrect password an HTTP error. I think in general _most_ form invalidations shouldn't be HTTP errors.